scene/main/window.cpp

```diff
@@ -1034,9 +1036,31 @@ bool Window::_can_consume_input_events() const {
 
 void Window::_window_input(const Ref<InputEvent> &p_ev) {
 	if (EngineDebugger::is_active()) {
-		// Quit from game window using F8.
+		// Quit from game window using the stop shortcut (F8 by default).
+		// The custom shortcut is provided via environment variable when running from the editor.
+		if (debugger_stop_shortcut.is_null()) {
+			String shortcut_str = OS::get_singleton()->get_environment("__GODOT_EDITOR_STOP_SHORTCUT__");
+			if (!shortcut_str.is_empty()) {
+				Variant shortcut_var;
+
+				VariantParser::StreamString ss;
+				ss.s = shortcut_str;
+
+				String errs;
+				int line;
+				VariantParser::parse(&ss, shortcut_var, errs, line);
+				debugger_stop_shortcut = shortcut_var;
+			}
+
+			if (debugger_stop_shortcut.is_null()) {
+				// Define a default shortcut if it wasn't provided or is invalid.
+				debugger_stop_shortcut.instantiate();
+				debugger_stop_shortcut->set_events({ (Variant)InputEventKey::create_reference(Key::F8) });
```

Comment:
Why is this being hardcoded again instead of using `ED_GET_SHORTCUT("editor/stop")`?

Reply:
Ah it's not editor code obviously :)

Reply:
Because ED_GET_SHORTCUT is only available in the editor and this is `scene`.

This is also the reason why it was completely hard-coded originally.